src/sink/ws.rs

```diff
@@ -214,123 +414,48 @@ impl offramp::Impl for Ws {
-                            let err_response = Self::create_error_response(&id, err.to_string());
-                            self.reply_tx
-                                .send(sink::Reply::Response(ERR, err_response))
-                                .await?;
-                        }
-                    },
-                    Err(err) => {
-                        error!("[Sink:{}] {}", self.sink_url, err);
-                        let err_response = Self::create_error_response(&id, err.to_string());
-                        self.reply_tx
-                            .send(sink::Reply::Response(ERR, err_response))
-                            .await?;
+                    if let Some((_, handle)) = self.connections.remove(&url) {
+                        handle.cancel().await;
                     }
-                },
+                }
             }
         }
         Ok(())
     }
 
-    fn build_response_events(&mut self, id: &Ids, msg: WsMessage) -> Result<Vec<Event>> {
-        let mut meta = Value::object_with_capacity(1);
-        let response_bytes = match msg {
-            WsMessage::Text(d) => {
-                meta.insert("binary", false)?;
-                d.into_bytes()
-            }
-            WsMessage::Binary(d) => d,
-        };
-
-        let mut ingest_ns = nanotime();
-        let preprocessed = preprocess(
-            &mut self.preprocessors,
-            &mut ingest_ns,
-            response_bytes,
-            &TremorURL::from_offramp_id("ws")?, // TODO: get proper url from offramp manager
-        )?;
-        let mut events = Vec::with_capacity(preprocessed.len());
-
-        for pp in preprocessed {
-            events.push(
-                LineValue::try_new(vec![pp], |mutd| {
-                    // ALLOW: mutd is vec![pp] in the line above
-                    let mut_data = mutd[0].as_mut_slice();
-                    let body = self
-                        .codec
-                        .decode(mut_data, nanotime())?
-                        .unwrap_or_else(Value::object);
-
-                    Ok(ValueAndMeta::from_parts(body, meta.clone())) // TODO: no need to clone the last element?
-                })
-                .map_err(|e: rental::RentalError<Error, _>| e.0)
-                .map(|data| Event {
-                    id: id.clone(),
-                    origin_uri: None, // TODO
-                    data,
-                    ..Event::default()
-                })?,
-            );
-        }
-        Ok(events)
-    }
-
-    fn create_error_response(event_id: &Ids, e: String) -> Event {
+    fn create_error_response(event_id: &Ids, e: &str, origin_uri: &EventOriginUri) -> Event {
         let mut error_data = simd_json::value::borrowed::Object::with_capacity(2);
-        error_data.insert_nocheck("error".into(), Value::from(e.clone()));
+        error_data.insert_nocheck("error".into(), Value::from(e.to_string()));
```

Comment:
Would `message` be a better name here? `error.error` might look a bit odd

Reply:
everywhere else we use `error`. I think we have an issue for standardizing error messages. Would you be fine doing it as part of that issue? --> https://github.com/tremor-rs/tremor-runtime/issues/506

Reply:
good call!